### PR TITLE
DAOS-16696 cart: Fix rc in error path

### PR DESCRIPTION
**Pull request by frostedcmos**
- Fix rc in error path during ivo_on_update failure

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/cart/crt_iv.c [hunk 1]
```diff
@@ -2911,8 +2911,11 @@ bulk_update_transfer_done_aux(const struct crt_bulk_cb_info *info)
 	return rc;
 
 send_error:
-	rc = crt_bulk_free(cb_info->buc_bulk_hdl);
 	output->rc = rc;
+	rc         = crt_bulk_free(cb_info->buc_bulk_hdl);
+	if (rc != 0)
+		DL_ERROR(rc, "crt_bulk_free() failed\n");
```
Comment (jolivier23) [inline src/cart/crt_iv.c:2917]: need to remove the \n because DL_ERROR adds it

Comment (jolivier23) [inline src/cart/crt_iv.c:2917]: Do you want to overwrite output->rc if it is 0?

**Review by jolivier23 — CHANGES_REQUESTED** [review]

File: src/cart/crt_iv.c (hunk 1 above)
Comment (frostedcmos) [reply src/cart/crt_iv.c:2917]: will do

Comment (frostedcmos) [reply src/cart/crt_iv.c:2917]: no, this is the 'send_error' path/label, and the rc returned from whatever failed above is the rc we want to send back. bulk_free() rc is only useful here for the subsequent DL_ERROR but nothing else

**Review by jolivier23 — APPROVED** [review]

**Review by soumagne — APPROVED** [review]

**Review by jgmoore-or — APPROVED** [review]

**Pull request merged**